io/include/gz/common/DataFrame.hh

```diff
@@ -179,7 +179,8 @@ namespace gz
         using FactoryT =
             math::InMemoryTimeVaryingVolumetricGridFactory<T, V>;
         std::vector<FactoryT> factories(dataColumnIndices.size());
-        for (auto it = _begin; it != _end; ++it)
+        auto it = _begin;
+        while (++it != _end)
```

Comment:
Knit: This works although it isn't immediately obvious what's going on. It may be cleaner to use `for (auto it = std::next(_begin); it != _end; ++it)` since we already check if `_begin` is valid before.

Reply:
@arjo129 that's way cleaner. Applied in https://github.com/gazebosim/gz-common/pull/435/commits/b478475964af847ef04aa6920883170051ca3974.